### DISCUSSION TITLE
USB Host HUB detection

### DISCUSSION THREAD

**Discussion by cecio** [adafruit/Adafruit_TinyUSB_Arduino#330] [Q&A] [answered]
Hello.

A quick question about the USB host functionalities of the TinyUSB Library:

if I well understood, the insertion of a USB HUB device is not going to call any callback function (like `tuh_mount_cb`). Is this correct? If so, is there a specific reason for this?

Thanks a lot!

**Comment by hathach** [comment, marked as answer]
right, currently there is no callback for hub. There is no specific reason, since hub is fully managed by tinyusb stack. Though if you have a good reason, I could change this behavior

**Comment by cecio** [comment]
First of all, thank you for the answer (and for your work on this software).

TBH I don't have a use case general enough, I'm exploring this as well and I see no other purpose than being able to identify all types of devices for "reporting" reasons, which is pretty specific to my case, I'm not sure it could be helpful for other use cases.

I'll have a look if it make sense to implement it into my code, leveraging tinyusb stack in some ways. If you have suggestions about which would be the best approach, they are welcome ;-)

Thanks again!
